iis-wf/iis-wf-citationmatching/src/main/java/eu/dnetlib/iis/wf/citationmatching/CitationMatchingReporter.java

```diff
@@ -0,0 +1,90 @@
+package eu.dnetlib.iis.wf.citationmatching;
+
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+
+import eu.dnetlib.iis.citationmatching.schemas.Citation;
+import eu.dnetlib.iis.common.schemas.ReportParam;
+import pl.edu.icm.sparkutils.avro.SparkAvroSaver;
+
+/**
+ * Reporter of citation matching job counters.<br/>
+ * It calculates citation matching counters and saves them
+ * as {@link ReportParam} datastore.
+ * 
+ * @author madryk
+ */
+public class CitationMatchingReporter {
```

Comment:
My comment related to `CitationMatchingDirectReporter` class name is applicable here as well. Since reporter is counter based it will be easier to understand what it does having `Counter` in class name.